example/test.py

```diff
@@ -41,24 +41,30 @@ def assert_tini_pid_1(ps_row, cmd):
                 MatchesStructure.byEquality(pid='1', ruser='root', args=args))
 
 
-def matches_attributes_values(attributes, values):
+def find_prefork_worker_split(ps_rows):
     """
-    Returns a matcher that matches the values of several attributes of an
-    object.
+    Given a list of PsRow objects for a group of processes using a pre-fork
+    worker process model, finds the row for the "master" process as well as
+    a list of "worker" processes.
     """
-    return MatchesStructure.byEquality(
-        **{a: v for a, v in zip(attributes, values)})
+    ppid_to_rows = {}
+    for row in ps_rows:
+        ppid_to_rows.setdefault(row.ppid, []).append(row)
 
+    # There should only be two ppids: the parent of the master and the master
+    assert len(ppid_to_rows) == 2
```

Comment:
Assertions like this aren't very helpful for debugging, because all we know from the failure is that we had a different number of entries from what we're expecting.